### PR TITLE
Naf

### PR DESCRIPTION
**Pull request by wadaniel**
fixed weight scaling, openmp naf, dqn, ddqn,
added mask for naf

### CODE REVIEW

**Review by SergioMartin86 — APPROVED** [review]

File: source/modules/neuralNetwork/neuralNetwork._cpp [hunk 1]
```diff
@@ -116,10 +118,6 @@ std::vector<float> NeuralNetwork::generateInitialHyperparameters()
   for (size_t i = 0; i < _layers.size(); i++)
   {
     auto layerParameters = _layers[i]->generateInitialHyperparameters();
-    // tmp fix (DW)
-    if (i == _layers.size() - 1)
-      for (size_t j = 0; j < layerParameters.size(); ++j)
-        layerParameters[j] *= 0.1;
```
Comment (webepasc) [inline source/modules/neuralNetwork/neuralNetwork._cpp:122]: are we still handling this somehow?

File: source/modules/solver/agent/continuous/DDPG/DDPG._cpp [hunk 2]
```diff
@@ -39,6 +38,9 @@ void DDPG::initializeAgent()
       _criticExperiment["Problem"]["Inputs"][i][j] = 0.0;
   }
 
+  size_t criticOutputLayerId = _criticNeuralNetwork["Layers"].size() - 1;
+  _criticNeuralNetwork["Solver"]["Neural Network"]["Layers"][criticOutputLayerId]["Weight Scaling"] = 0.1;
```
Comment (webepasc) [inline source/modules/solver/agent/continuous/DDPG/DDPG._cpp:42]: aha, yes, apparently

**Review by webepasc — APPROVED** [review]
Looks ok

File: source/modules/solver/agent/continuous/DDPG/DDPG._cpp (hunk 2 above)
Comment (wadaniel) [reply source/modules/solver/agent/continuous/DDPG/DDPG._cpp:42]: yes, I added it to all continuous agents

**Pull request merged**